### PR TITLE
Feat/keyp12 105 unit tests potransaction

### PR DESCRIPTION
**Pull request by jeanpierrefortune**

### CODE REVIEW

File: java/component/keyple-calypso/src/main/java/org/eclipse/keyple/calypso/transaction/PoTransaction.java [hunk 1]
```diff
@@ -71,11 +71,12 @@
      * to the length of the outgoing data plus 6 bytes
      */
     private static final int SESSION_BUFFER_CMD_ADDITIONAL_COST = 6;
+    private static final int APDU_HEADER_LENGTH = 5;
 
     /** Ratification command APDU for rev <= 2.4 */
-    private static final byte[] RATIFICATION_CMD_APDU_LEGACY = ByteArrayUtil.fromHex("94B2000000");
+    private static final byte[] RATIFICATION_CMD_APDU_LEGACY = ByteArrayUtil.fromHex("94B2010000");
```
Comment (pterr) [inline java/component/keyple-calypso/src/main/java/org/eclipse/keyple/calypso/transaction/PoTransaction.java:77]: We may extract RATIFICATION_CMD_APDU_LEGACY & RATIFICATION_CMD_APDU and define them in a public interface.
And in PoSecuritySettings, we could maybe propose a way to personalize the ratification cmd.

File: java/component/keyple-calypso/src/main/java/org/eclipse/keyple/calypso/transaction/PoTransaction.java [hunk 2]
```diff
@@ -947,7 +969,8 @@ public final void processClosing(ChannelControl channelControl)
         }
 
         // Finally, close the session as requested
-        processAtomicClosing(poAtomicCommands, calypsoPo.getTransmissionMode(), channelControl);
+        processAtomicClosing(poAtomicCommands, poSecuritySettings.getRatificationMode(),
```
Comment (pterr) [inline java/component/keyple-calypso/src/main/java/org/eclipse/keyple/calypso/transaction/PoTransaction.java:972]: poSecuritySettings.getRatificationMode() is optionnal

Comment (jeanpierrefortune) [reply java/component/keyple-calypso/src/main/java/org/eclipse/keyple/calypso/transaction/PoTransaction.java:972]: Yes, processAtomicClosing is private.
processClosing has no ratification mode argument.
The default value for the ratification mode is defined in PoSecuritySettings.
We use the possibility to call processAtomicClosing with CLOSE_RATIFIED when doing multiple sessions.

File: java/component/keyple-calypso/src/main/java/org/eclipse/keyple/calypso/transaction/PoTransaction.java (hunk 1 above)
Comment (jeanpierrefortune) [reply java/component/keyple-calypso/src/main/java/org/eclipse/keyple/calypso/transaction/PoTransaction.java:77]: The Calypso requirements define 00B2010000 as ratification command. Do we need to propose a way to define something different?

Comment (jeanpierrefortune) [reply java/component/keyple-calypso/src/main/java/org/eclipse/keyple/calypso/transaction/PoTransaction.java:77]: Anyway, we need to improve the ratification command building.

**Pull request merged**